src/main/java/org/spongepowered/mod/mixin/entity/MixinEntityItem.java

```diff
@@ -0,0 +1,170 @@
+
+    //
+    // In the case where a Forge mod sets the delay to MAGIC_INFINITE_PICKUP_DELAY, but a plugin has
+    // never called setPickupDelay or setInfinitePickupDelay, delayBeforeCanPickup would be decremented,
+    // as infiniteDelay is set to false. However, this is not the intended behavior, as the Forge
+    // mod meant an infinite delay to be set.
+
+    // To resolve the ambiguity, this flag is used to determine whether infiniteDelay is false because it was never changed
+    // from the default, or if it was explicitly set by a plugin
+    private boolean pluginPickupSet;
+
+    private boolean infinitePickupDelay;
+
+    private boolean pluginDespawnSet;
+
+    private boolean infiniteDespawnDelay;
+
+    public MixinEntityItem(World worldIn) {
+        super(worldIn);
+    }
+
+    @Inject(method = "onUpdate()V", at = @At(value = "FIELD", target = "Lnet/minecraft/entity/item/EntityItem;delayBeforeCanPickup:I", opcode = Opcodes.PUTFIELD, shift = At.Shift.AFTER))
+    private void onOnUpdate(CallbackInfo ci) {
+        if (this.delayBeforeCanPickup == MAGIC_INFINITE_PICKUP_DELAY && !infinitePickupDelay && pluginPickupSet) {
+            this.delayBeforeCanPickup--;
+        }
+    }
+
+    @Inject(method = "onUpdate()V", at = @At(value = "FIELD", target = "Lnet/minecraft/entity/item/EntityItem;age:I", opcode = Opcodes.PUTFIELD, shift = At.Shift.AFTER))
+    private void onOnUpdateAge(CallbackInfo ci) {
+        if (this.delayBeforeCanPickup == MAGIC_INFINITE_DESPAWN_TIME && !infiniteDespawnDelay && pluginDespawnSet) {
+            this.delayBeforeCanPickup--;
+        }
+    }
+
+    @Override
+    public int getPickupDelay() {
+        if (this.delayBeforeCanPickup == MAGIC_INFINITE_PICKUP_DELAY) {
+            // There are two cases when -1 should be returned:
+
+            // The plugin has called set an infinite pickup delay
+            // The plugin has not set a pickup delay (neither setPickupDelay nor setInfinitePickupDelay
+            // has been called) - a Forge mod or something else has set the pickup delay, and they presumably
+            // know about the magic value.
+            if ((this.pluginPickupSet && this.infinitePickupDelay) || !this.pluginPickupSet) {
+                return -1;
+            }
+        }
+        return this.delayBeforeCanPickup;
+    }
+
+    @Override
+    public void setPickupDelay(int delay) {
+        this.delayBeforeCanPickup = delay;
+        this.pluginPickupSet = true;
+        this.infinitePickupDelay = false;
+    }
+
+    @Override
+    public void setInfinitePickupDelay() {
+        this.delayBeforeCanPickup = MAGIC_INFINITE_PICKUP_DELAY;
+        this.pluginPickupSet = true;
+        this.infinitePickupDelay = true;
+    }
+
+    @Override
+    public int getDespawnTime() {
+        if (this.age == MAGIC_INFINITE_DESPAWN_TIME) {
+            if ((this.pluginDespawnSet && this.infiniteDespawnDelay) || !this.pluginDespawnSet) {
+                return -1;
```

Comment:
Given that you have consts for the other magic numbers, would it not have a pleasing symmetry to also declare these two occurrences of -1 as such?